### PR TITLE
search redesign: sticky search sidebar that scrolls with content

### LINKED ISSUE #20596: Search sidebar and search results should scroll together
Sidebar should not scroll independently of search results.

### PR DESCRIPTION
**Pull request by limitedmage**
- Search sidebar now scrolls with content instead of being scrollable on its own
- Sidebar is now sticky using [react-sticky-box](https://react-sticky-box.codecks.io/)

Fixes #20596

https://user-images.githubusercontent.com/206864/117067978-8ab90d80-acdf-11eb-9679-9ac5c5189a87.mov

### CODE REVIEW

File: client/web/src/search/results/streaming/sidebar/SearchSidebar.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React, { useCallback } from 'react'
 import { useHistory } from 'react-router'
+import StickyBox from 'react-sticky-box'
```
Comment (5h1rU) [inline client/web/src/search/results/streaming/sidebar/SearchSidebar.tsx:3]: Any reason to use this instead of` position: sticky` in CSS?

Comment (limitedmage) [reply client/web/src/search/results/streaming/sidebar/SearchSidebar.tsx:3]: Unfortunately `position:sticky` alone doesn't work if the sidebar is taller than the viewport, which why JS is needed. The library internally uses `position:sticky` but listens to scroll and resize events to modify the positioning during scrolling.

**Review by 5h1rU — APPROVED** [review]
LGTM 💪

**Review by rrhyne — APPROVED** [review]
Works really well. Thanks!

**Review by novoselrok — APPROVED** [review]

**Pull request merged**